core/storage/btree.rs

```diff
@@ -909,7 +946,7 @@ impl BTreeCursor {
             *remaining_to_read -= to_read;
 
             if *remaining_to_read != 0 && next != 0 {
-                let (new_page, c) = self.pager.read_page(next as i64)?;
+                let (new_page, c) = return_if_io!(self.pager.read_page(next as i64));
```

Comment:
not re-entrant. We are already returning a completion here, is `return_if_io!` necessary?